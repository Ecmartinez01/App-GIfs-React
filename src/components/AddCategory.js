import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategory}) => {
    const [texto , setTexto ] = useState("")

    const encuentraTexto = (e)=>{
        setTexto(e.target.value)
    }
    const submit = (e)=>{
        e.preventDefault()
        console.log("Submit")
        if (texto.trim().length > 2) {
            setCategory(cats => [texto,...cats])
            setTexto('')
        }
    }
    return (
        <form onSubmit={submit} >
            <p> {texto}</p>
            <input
            value={texto}
            type="text"
            onChange={encuentraTexto}
            />
        </form>
        
    )
}
AddCategory.propTypes ={
    setCategory : PropTypes.func.isRequired       
}