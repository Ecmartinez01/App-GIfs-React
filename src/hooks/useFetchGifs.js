import { useEffect, useState } from "react"
import { getGif } from '../helpers/getGif';
export const useFetchGif =(categoria)=>{
    const [state, setState] = useState({
        data:[],
        loading:true
    })
    useEffect(()=>{
        
        getGif(categoria).then(gifs =>{
           
                setState({
                    data:gifs,
                    loading:false
                })
                
            ;
        })
    },[categoria])
   
    return state
}