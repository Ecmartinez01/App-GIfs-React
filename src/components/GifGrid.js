import React /*{ useEffect, useState }*/ from 'react';
import { useFetchGif } from '../hooks/useFetchGifs';
// import { getGif } from '../helpers/getGif';
import { GifGridItem } from './GifGridItem';
export const GifGrid =({categoria})=>{

      const {data,loading} = useFetchGif(categoria)

    
    return(<>
        <h3 className='animate__animated animate__flip animate__delay-0.5s '>{categoria}</h3>
        {loading && <p className='animate__animated animate__bounceOut animate__delay-1s '>Cargando</p>}
        <div className='card-grid'>
            {data.map(gif => (
               <GifGridItem 
               key={gif.id}
               {...gif}
               /> 
                
            ))}
      
        </div>
    </>
    )
}