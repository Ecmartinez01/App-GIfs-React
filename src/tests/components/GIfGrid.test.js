import React from 'react';
import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid";
import  {useFetchGif}  from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas con GifGrid ', () => {
    const categoria = "soccer"

    it('Debe mostrar correctamente el componente', () => {
        useFetchGif.mockReturnValue({
          data:[],
          loading:true  
        })
        const wrapper = shallow(<GifGrid categoria={categoria}/>)
        expect(wrapper).toMatchSnapshot()
     })
    it('Deben de mostrar items cuando se carguen imagenes con nuestri useFet hc', () => {
        const gifs = []
        useFetchGif.mockReturnValue({
            data:gifs,
            loading:false  
          })
        const wrapper = shallow(<GifGrid categoria={categoria}/>)
        expect(wrapper.find('p').exists()).toBe(false)
    })
    
})
