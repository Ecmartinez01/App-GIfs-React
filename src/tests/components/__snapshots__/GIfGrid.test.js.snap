// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifGrid  Debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flip animate__delay-0.5s "
  >
    soccer
  </h3>
  <p
    className="animate__animated animate__bounceOut animate__delay-1s "
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
